x-pack/plugins/stack_alerts/public/alert_types/threshold/expression.tsx

```diff
@@ -133,6 +142,13 @@ export const IndexThresholdAlertTypeExpression: React.FunctionComponent<
     setEsFields(currentEsFields);
   };
 
+  const handleFilterChange = useCallback(
```

Comment:
is there any way to add a debounce here so not every keystroke is validated?

Reply:
Debouncing here just prevents the value to be set.
Current validation is triggered on every field change of a rule. We need to change the whole validation implementation on framework level in order to tackle validations for each field separately.